recipes/full_finetune_distributed.py

```diff
@@ -913,8 +913,19 @@ def train(self) -> None:
                         torch.distributed.all_reduce(num_tokens)
                         # This will ensure that the logged loss matches what we're optimizing
                         torch.distributed.all_reduce(running_loss)
+
                         # Manually scale the gradients from unnormalized loss by total # of tokens
-                        training.scale_grads(self._model, self.dp_degree / num_tokens)
+                        def scale_grads_fn():
```

Comment:
Dumb q: Why can't we compile the function directly instead of having this extra function?

Reply:
Also, will this be addressed in this PR?

```Current repo compilation fails if to set skip_rope_interval=4,, have to test with skip_rope_interval=None,```

Reply:
> skip_rope_interval

There is a problem in flex_attention mask_mod - "Cuda illegal memory access" - some bad calculation of indices.
This needs a separate fix, will ask Flex attention engineers to take a look at a minimized repro.

Reply:
> I have not tried compiling object.method before. I can try.

@IvanKobzarev Did this work?

Reply:
Separately I have a similar question to the one I left on the compiled optimizer step PR -- do we need to do this every step? If not I wonder if we could just do something like `self.grad_scaler = get_grad_scale_fn(self._compile)` once during init, returning the maybe-compiled scale function, then call it in a single line here.